src/map/status.c

```diff
@@ -8723,40 +8723,186 @@ static int status_change_start_sub(struct block_list *src, struct block_list *bl
 				//break;
 				}
-			case SC_ARMOR_RESIST:
-				if (val1 > 0) {
+			case SC_ARMOR_RESIST: {
+				struct status_change_entry *sce_water = sc->data[SC_RESIST_PROPERTY_WATER];
+				struct status_change_entry *sce_ground = sc->data[SC_RESIST_PROPERTY_GROUND];
+				struct status_change_entry *sce_fire = sc->data[SC_RESIST_PROPERTY_FIRE];
+				struct status_change_entry *sce_wind = sc->data[SC_RESIST_PROPERTY_WIND];
+
+				// Water
+				int sum_water = val1 + ((sce_fire != NULL) ? sce_fire->val2 : 0);
+				bool show_icon = true;
+				if (sce_water != NULL) {
+					sum_water += sce_water->val1;
+					int left = (int)DIFF_TICK(timer->get(sce_water->timer)->tick, timer->gettick());
+					if (left > total_tick && sum_water - val1 > 0)
+						show_icon = false;
+				}
+				if (val1 > 0 && sum_water > 0 && show_icon) {
 					int sc_icn = status->get_sc_icon(SC_RESIST_PROPERTY_WATER);
 					int sc_typ = status->get_sc_relevant_bl_types(SC_RESIST_PROPERTY_WATER);
 					clif->status_change(bl, sc_icn, sc_typ, 1, total_tick, 0, 0, 0);
+				} else if (sum_water <= 0) {
+					clif->sc_end(bl, bl->id, AREA, status->get_sc_icon(SC_RESIST_PROPERTY_WATER));
 				}
 
-				if (val2 > 0) {
+				// Ground
+				int sum_ground = val2 + ((sce_wind != NULL) ? sce_wind->val2 : 0);
+				show_icon = true;
+				if (sce_ground != NULL) {
+					sum_ground += sce_ground->val1;
+					int left = (int)DIFF_TICK(timer->get(sce_ground->timer)->tick, timer->gettick());
+					if (left > total_tick && sum_ground - val2 > 0)
+						show_icon = false;
+				}
+				if (val2 > 0 && sum_ground > 0 && show_icon) {
 					int sc_icn = status->get_sc_icon(SC_RESIST_PROPERTY_GROUND);
 					int sc_typ = status->get_sc_relevant_bl_types(SC_RESIST_PROPERTY_GROUND);
 					clif->status_change(bl, sc_icn, sc_typ, 1, total_tick, 0, 0, 0);
+				} else if (sum_ground <= 0) {
+					clif->sc_end(bl, bl->id, AREA, status->get_sc_icon(SC_RESIST_PROPERTY_GROUND));
 				}
 
-				if (val3 > 0) {
+				// Fire
+				int sum_fire = val3 + ((sce_ground != NULL) ? sce_ground->val2 : 0);
+				show_icon = true;
+				if (sce_fire != NULL) {
+					sum_fire += sce_fire->val1;
+					int left = (int)DIFF_TICK(timer->get(sce_fire->timer)->tick, timer->gettick());
+					if (left > total_tick && sum_fire - val3 > 0)
+						show_icon = false;
+				}
+				if (val3 > 0 && sum_fire > 0 && show_icon) {
 					int sc_icn = status->get_sc_icon(SC_RESIST_PROPERTY_FIRE);
 					int sc_typ = status->get_sc_relevant_bl_types(SC_RESIST_PROPERTY_FIRE);
 					clif->status_change(bl, sc_icn, sc_typ, 1, total_tick, 0, 0, 0);
+				} else if (sum_fire <= 0) {
+					clif->sc_end(bl, bl->id, AREA, status->get_sc_icon(SC_RESIST_PROPERTY_FIRE));
 				}
 
-				if (val4 > 0) {
+				// Wind
+				int sum_wind = val4 + ((sce_water != NULL) ? sce_water->val2 : 0);
+				show_icon = true;
+				if (sce_wind != NULL) {
+					sum_wind += sce_wind->val1;
+					int left = (int)DIFF_TICK(timer->get(sce_wind->timer)->tick, timer->gettick());
```

Comment:
``timer->get`` may return NULL. also sce_wind->timer can be invalid timer. probably need add checks for timer field and return value of ``timer->get``?